### PR TITLE
b/166158386 #3: Add ApiKeyState into report log entry

### PR DESCRIPTION
**Pull request by nareddyt**
This will make analysis of requests with API Keys easier in the future. Log entries will contain 1 of the 4 states now:
- NOT CHECKED
- VERIFIED
- NOT ENABLED
- INVALID

Testing done: Unit and integration tests.

Follow-up PR: Consider adding a new state `MISSING`. If a method requires an API Key but one is not provided, we consider that `NOT_CHECKED` now. While that's technically correct (we don't make a CHECK call), to the API Producer that's a little misleading. This is low priority though.

### CODE REVIEW

File: tests/utils/service_control_utils_test.go [hunk 1]
```diff
@@ -445,6 +445,12 @@ operations: <
         value: <
           string_value: "api-key"
         >
+      >
+			fields: <
```
Comment (qiwzhang) [inline tests/utils/service_control_utils_test.go:449]: Please fix the indent.

Comment (nareddyt) [reply tests/utils/service_control_utils_test.go]: Oops, tabs instead of spaces... Done

**Pull request merged**